src/main.rs

```diff
@@ -530,75 +518,85 @@ fn cmd_inspect(
 
-        return Ok(());
-    }
+        if sub_args.mode == FetchMode::Sourcegraph {
+            let url = format!("https://sourcegraph.com/crates/{package}@v{version}");
+            tokio::runtime::Handle::current()
+                .block_on(prompt_criteria_eulas(
+                    out,
+                    cfg,
+                    network.as_ref(),
+                    &store,
+                    package,
+                    None,
+                    version,
+                    Some(&url),
+                ))
+                .into_diagnostic()?;
+
+            open::that(&url).into_diagnostic().wrap_err_with(|| {
+                format!("Couldn't open {url} in your browser, try --mode=local?")
+            })?;
+
+            writeln!(out, "\nUse |cargo vet certify| to record your audit.");
+            return Ok(());
+        }
 
-    let fetched = tokio::runtime::Handle::current().block_on(async {
-        let (pkg, eulas) = tokio::join!(
-            cache.fetch_package(network.as_ref(), package, version),
-            prompt_criteria_eulas(
-                out,
-                cfg,
-                network.as_ref(),
-                &store,
-                package,
-                None,
-                version,
-                None,
-            ),
-        );
-        eulas.into_diagnostic()?;
-        pkg.into_diagnostic()
-    })?;
+        tokio::runtime::Handle::current().block_on(async {
+            let (pkg, eulas) = tokio::join!(
+                cache.fetch_package(network.as_ref(), package, version),
+                prompt_criteria_eulas(
+                    out,
+                    cfg,
+                    network.as_ref(),
+                    &store,
+                    package,
+                    None,
+                    version,
+                    None,
+                ),
+            );
+            eulas.into_diagnostic()?;
+            pkg.into_diagnostic()
+        })?
+    };
 
     #[cfg(target_family = "unix")]
-    {
+    if let Some(shell) = std::env::var_os("SHELL") {
         // Loosely borrowed from cargo crev.
-        let shell = std::env::var_os("SHELL").unwrap();
         writeln!(out, "Opening nested shell in: {:#?}", fetched);
         writeln!(out, "Use `exit` or Ctrl-D to finish.",);
-        let mut command = std::process::Command::new(shell);
-        command.current_dir(fetched.clone()).env("PWD", fetched);
-        panic_any(ExecPanic(command));
-    }
+        let status = std::process::Command::new(shell)
+            .current_dir(fetched.clone())
+            .env("PWD", fetched)
+            .status()
+            .map_err(CommandError::CommandFailed)
+            .into_diagnostic()?;
 
-    #[cfg(not(target_family = "unix"))]
-    {
-        writeln!(out, "  fetched to {:#?}", fetched);
-        Ok(())
+        writeln!(out, "\nUse |cargo vet certify| to record your audit.");
+
+        if let Some(code) = status.code() {
+            panic_any(ExitPanic(code));
```

Comment:
oh rip lol. we'll get 'em eventually.

I'm assuming I can do something like have a ExitError and dynamic cast it out of the Report...